lib/node/source.js

```diff
@@ -58,22 +58,27 @@ class StreamSource {
     this.size = null;
 
     stream.pause();
+    this._done = false;
+    stream.on("end", () => this._done = true);
 
     this._buf = new Buffer(chunkSize);
     this._bufPos = null;
     this._bufLen = 0;
   }
 
+  _copyBuf(start, end) {
```

Comment:
I wouldn't call this function _copyBuf because Buffer#slice does not create a copy of the buffer's data. Quoting the relevant section from the docs (https://nodejs.org/dist/latest-v10.x/docs/api/buffer.html#buffer_buf_slice_start_end):

> Returns a new Buffer that references the same memory as the original, but offset and cropped by the start and end indices.
[...]
Modifying the new Buffer slice will modify the memory in the original Buffer because the allocated memory of the two objects overlap.

It's not relevant for our situation as we don't modify any buffers' content. But calling this function _copyBuf even if it just creates a new reference to the same buffer data is rather confusing.